stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/BallerinaHTTPConnectorListener.java

```diff
@@ -44,22 +47,47 @@ public BallerinaHTTPConnectorListener(HTTPServicesRegistry httpServicesRegistry)
 
     @Override
     public void onMessage(HTTPCarbonMessage httpCarbonMessage) {
-        HttpResource httpResource = HttpDispatcher.findResource(httpServicesRegistry, httpCarbonMessage);
+        HttpResource httpResource;
+        if (!isAccessed(httpCarbonMessage)) {
+            httpResource = HttpDispatcher.findResource(httpServicesRegistry, httpCarbonMessage);
+            if (HttpDispatcher.isDiffered(httpResource)) {
+                httpCarbonMessage.setProperty(HTTP_RESOURCE, httpResource);
+            } else {
+                extractPropertiesAndStartResourceExecution(httpCarbonMessage, httpResource);
+            }
+        } else {
+            httpResource = (HttpResource) httpCarbonMessage.getProperty(HTTP_RESOURCE);
+            extractPropertiesAndStartResourceExecution(httpCarbonMessage, httpResource);
+        }
+    }
+
+    @Override
+    public void onError(Throwable throwable) {
+        log.error("Error in http server connector", throwable);
```

Comment:
log error msg as well?
log.error("Error in http server connector" + throwable.getMessage(), throwable);

Reply:
Ack